### PR TITLE
Predictive tuning improvements

### PR DESCRIPTION
**Pull request by Jalle19**
@ksooo this is still WIP (the current commit works though) but you might wanna take a look.

### CODE REVIEW

**Comment by Jalle19** [comment]
Fixes https://github.com/kodi-pvr/pvr.hts/issues/115 by the way, remind me to add it to the commit message too.

**Comment by Glenn-1990** [comment]
I also spotted another bug with predictive tuning...
https://github.com/kodi-pvr/pvr.hts/blob/master/src/Tvheadend.cpp#L2026
The weight of the new active subscription is set to normal before! the old subscription gets decreased in weight. So the 2 subscriptions will be set to normal at the same time for a very short period, this could lead to hiccups in other streams if no other tuners are available.

I can also make a PR for this?

**Comment by Jalle19** [comment]
Nice catch, fixing that would probably fix #91. If you fix I can pick your commit into this PR, no need to make a separate pull.

**Comment by Jalle19** [comment]
@Glenn-1990 don't bother with a fix, I'm working on this right now anyway so I'll add a commit for it. Thanks for bringing it to attention though!

**Comment by Jalle19** [comment]
@ksooo no longer WIP now.

**Comment by ksooo** [comment]
Stumbled only over the magic number used by the new channel predictor. Everything else looks good to me.

**Comment by Jalle19** [comment]
@ksooo added `CHANNEL_ID_NONE` to replace the magic number.

**Comment by Jalle19** [comment]
@ksooo good to go?

File: src/tvheadend/ChannelTuningPredictor.h [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+{
+
+  namespace predictivetune
+  {
+
+    /**
+     * Used to indicate that predictive tuning failed to determine which
+     * channel to be tuned next
+     */
+    const unsigned int CHANNEL_ID_NONE = -1;
```
Comment (ksooo) [inline src/tvheadend/ChannelTuningPredictor.h]: unsigned ... = -1 ? Really?

Comment (Jalle19) [inline src/tvheadend/ChannelTuningPredictor.h]: Yes, it's the same as UINT32_MAX

Comment (ksooo) [inline src/tvheadend/ChannelTuningPredictor.h]: Ofc, just wanted to be sure that this was intended.

One last thing: why unsigned int and not uint32_t like used elsewhere with channel ids? https://github.com/kodi-pvr/pvr.hts/pull/117/files#diff-ce6ea06b04dac3a0c1f703fe9ed2cc37R65

Comment (Jalle19) [inline src/tvheadend/ChannelTuningPredictor.h]: Just a habit, I can change it before merging.

**Comment by ksooo** [comment]
good to go.

**Pull request merged**